hamocc/mo_carchm.F90

```diff
@@ -0,0 +1,880 @@
+                              isco212,isilica, &
+                              iatmf11,iatmf12,iatmsf6,icfc11,icfc12,isf6, &
+                              iatmc13,iatmc14,icalc13,icalc14,idet14,idoc14,iphy14,isco213,isco214,izoo14,safediv, &
+                              iatmnco2,inatalkali,inatcalc,inatsco212, &
+                              ks,issso14,isssc14,ipowc14, &
+                              iatmbromo,ibromo
+    use mo_param_bgc,   only: c14dec,atm_co2_nat
+    use mo_vgrid,       only: dp_min,kmle,kbo,ptiestu
+    use mo_carbch,      only: atm_cfc11_nh,atm_cfc11_sh,atm_cfc12_nh,atm_cfc12_sh,atm_sf6_nh,atm_sf6_sh, &
+                              co213fxd,co213fxu,co214fxd,co214fxu, &
+                              nathi,natco3,natpco2d,natomegaa,natomegac
+    use mo_sedmnt,      only: sedlay,powtra,burial
+
+    ! Arguments
+    integer, intent(in) :: kpie                                              !  1st dimension of model grid.
+    integer, intent(in) :: kpje                                              !  2nd dimension of model grid.
+    integer, intent(in) :: kpke                                              !  3rd (vertical) dimension of model grid.
+    integer, intent(in) :: kbnd                                              !  nb of halo grid points
+    real,    intent(in) :: pdlxp(kpie,kpje)                                  !  size of scalar grid cell (1st dimension) [m].
+    real,    intent(in) :: pdlyp(kpie,kpje)                                  !  size of scalar grid cell (2nd dimension) [m].
+    real,    intent(in) :: pddpo(kpie,kpje,kpke)                             !  size of scalar grid cell (3rd dimension) [m].
+    real,    intent(in) :: prho(kpie,kpje,kpke)                              !  density [g/cm^3].
+    real,    intent(in) :: pglat(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)          !  latitude of grid cells [deg north].
+    real,    intent(in) :: omask(kpie,kpje)                                  !  ocean mask.
+    real,    intent(in) :: psicomo(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)        !  sea ice.
+    real,    intent(in) :: ppao(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)           !  sea level presure [pascal].
+    real,    intent(in) :: pfu10(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd)          !  forcing field wind speed.
+    real,    intent(in) :: ptho(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd,kpke)      !  potential temperature.
+    real,    intent(in) :: psao(1-kbnd:kpie+kbnd,1-kbnd:kpje+kbnd,kpke)      !  salinity [psu].
+
+    ! Local variables
+    integer, parameter :: niter=20
+    integer :: i,j,k,l,js
+    real    :: supsat, undsa, dissol
+    real    :: rpp0,fluxd,fluxu
+    real    :: kwco2,kwo2,kwn2,kwdms,kwn2o
+    real    :: scco2,sco2,scn2,scdms,scn2o
+    real    :: xconvxa
+    real    :: oxflux,niflux,dmsflux,n2oflux
+    real    :: ato2,atn2,atco2,pco2
+    real    :: oxy,ani,anisa
+    real    :: rrho,t,t2,t3,t4,tk,tk100,prb,s,rs
+    real    :: kh,khd,k1,k2,kb,k1p,k2p,k3p,ksi,kw,ks1,kf,kspc,kspa
```

Comment:
Not sure, but also here the capital letters had a meaning (in my opinion)

Reply:
Yes I agree, generally capital letters in variable names have been there for a good reason and should stay.

Reply:
Right. Fixed.